src/pages/MunicipalitiesRankedPage.tsx

```diff
@@ -92,18 +104,47 @@ export function MunicipalitiesRankedPage() {
   const renderMapOrList = (isMobile: boolean) =>
     viewMode === "map" ? (
       <div className={isMobile ? "relative h-[65vh]" : "relative h-full"}>
-        <SwedenMap
+        <MapOfSweden
           geoData={geoData as FeatureCollection}
-          municipalityData={municipalities}
-          selectedKPI={selectedKPI}
-          onMunicipalityClick={handleMunicipalityClick}
+          data={municipalities.map((m) => {
+            const { sectorEmissions, ...rest } = m;
+            return { ...rest, id: m.name };
+          })}
+          selectedAttribute={selectedKPI}
+          onRegionClick={handleMunicipalityNameClick}
         />
       </div>
     ) : (
-      <MunicipalityRankedList
-        municipalityData={municipalities}
-        selectedKPI={selectedKPI}
-        onMunicipalityClick={handleMunicipalityClick}
+      <RankedList
+        data={municipalities}
+        selectedDataPoint={asDataPoint({
+          label: selectedKPI.label,
+          key: selectedKPI.key as keyof Municipality,
+          unit: selectedKPI.unit,
+          description: selectedKPI.description,
+          higherIsBetter: selectedKPI.higherIsBetter,
+          nullValues: selectedKPI.nullValues,
+          formatter: (value: unknown) => {
```

Comment:
I wonder if having this logic in the rankedList would be better than in the props 🤔 , or maybe just move it up in the file and pass as a var here